src/utils/evm/topUp.ts

```diff
@@ -0,0 +1,66 @@
+import { BigNumber, Wallet, providers, utils } from 'ethers'
+import { BalanceConfig, RpcEndpoints } from '../../types'
+import { EVM_BLOCK_CONFIRMATIONS } from '../../constants'
+
+export async function topUpEVMBalance(
+  balanceConfig: Array<BalanceConfig>,
+  rpcEndpoints: RpcEndpoints,
+  wallet: Wallet
+): Promise<void> {
+  await Promise.all(
+    balanceConfig.map(async (domain) => {
+      const provider = new providers.JsonRpcProvider(
+        rpcEndpoints[domain.chainId]
+      )
+      const signer = wallet.connect(provider)
+
+      for (const relayer of domain.nativeBalanceData) {
+        try {
+          console.log(
+            `Initiated relayer ${relayer.address} top up on network with chainId ${domain.chainId}`
+          )
+          const relayerBalance = await checkRelayerBalance(provider, relayer)
```

Comment:
```suggestion
          const relayerBalance = await getRelayerBalance(provider, relayer)
```

checkRelayerBalance implies it will error if it is lower.
This just fetches it.